spec/integration/post_spec.rb

```diff
@@ -0,0 +1,118 @@
+                          bio: 'Micronaut', post_counter: 0)
+      @first_post = Post.create(author: @user, title: 'Post number one', text: 'Post number one, this is the first one',
+                                likes_counter: 0, comments_counter: 0)
+      @second_post = Post.create(author: @user, title: 'Post number two',
+                                 text: 'Second post', likes_counter: 0, comments_counter: 0)
+      @third_post = Post.create(author: @user, title: 'Post number three',
+                                text: 'Third post', likes_counter: 0, comments_counter: 0)
+      @fourth_post = Post.create(author: @user, title: 'Post number four',
+                                 text: 'Fourth post', likes_counter: 0, comments_counter: 0)
+      @first_comment = Comment.create(author: @user, post: @second_post, text: 'Awesome post')
+      @second_comment = Comment.create(author: @user, post: @second_post, text: 'I like this post')
+      @third_comment = Comment.create(author: @user, post: @first_post, text: 'What a great post')
+      @fourth_comment = Comment.create(author: @user, post: @first_post, text: 'My favorite post')
+      @fifth_comment = Comment.create(author: @user, post: @first_post, text: 'Cool post')
+
+      visit user_posts_path(@user)
+    end
+
+    it "renders user's profile picture" do
+      find("img[src='#{@user.photo}']")
+    end
+
+    it "renders the user's username" do
+      expect(page).to have_content(@user.name)
+    end
+
+    it "renders the user's post count" do
+      expect(page).to have_content(@user.post_counter)
+    end
+
+    it "renders the user's posts" do
+      expect(page).to have_content(@fourth_post.title)
+      expect(page).to have_content(@second_post.title)
+      expect(page).to have_content(@third_post.title)
+    end
+
+    it "renders some of the post's body" do
+      expect(page).to have_content(@fourth_post.text[0, 50])
+      expect(page).to have_content(@second_post.text[0, 50])
+      expect(page).to have_content(@third_post.text[0, 50])
+    end
+
+    it 'renders first comments on a post' do
+      expect(page).to have_content(@first_comment.text)
+      expect(page).to have_content(@second_comment.text)
+    end
+
+    it 'renders comments count of a post' do
+      expect(page).to have_content(@fourth_post.comments_counter)
+      expect(page).to have_content(@second_post.comments_counter)
+      expect(page).to have_content(@third_post.comments_counter)
+    end
+
+    it 'renders likes count of a post' do
+      expect(page).to have_content(@fourth_post.likes_counter)
+      expect(page).to have_content(@second_post.likes_counter)
+      expect(page).to have_content(@third_post.likes_counter)
+    end
+
+    it "redirects to a specific post's show page" do
+      click_link @second_post.title
+      expect(page).to have_current_path(user_post_path(@user, @second_post))
+    end
+  end
```

Comment:
- [ ] [REQUIRED] Please take a look at the featured tests for `User` **post index page**: You have missed out on this test:
- Tests that assert you can see a section for pagination if there are more posts than fit on the view.

Kindly fix this before asking for another review.